modules/javafx.graphics/src/main/java/com/sun/javafx/stage/EmbeddedWindow.java

```diff
@@ -85,4 +88,13 @@ private void doVisibleChanging(boolean visible) {
         }
     }
 
+    public void setNodeOrientation(NodeOrientation nor) {
+        orientation = nor;
+        final Scene sceneValue = getScene();
+        SceneHelper.parentEffectiveOrientationInvalidated(sceneValue);
```

Comment:
You need to check whether the value has actually changed before calling invalidated or this will cause a performance regression.

Minor: I might not bother saving `getScene()` in a local var since you only use it once and `getScene()` is short.